src/main/lens-proxy.ts

```diff
@@ -73,12 +71,41 @@ export class LensProxy {
     if (cluster) {
       const proxyUrl = await cluster.contextHandler.resolveAuthProxyUrl() + req.url.replace(apiKubePrefix, "")
       const apiUrl = url.parse(cluster.apiUrl)
-      const res = new http.ServerResponse(req)
-      res.assignSocket(socket)
-      res.setHeader("Location", proxyUrl)
-      res.setHeader("Host", apiUrl.hostname)
-      res.statusCode = 302
-      res.end()
+      const pUrl = url.parse(proxyUrl)
+      const connectOpts = {port: parseInt(pUrl.port), host: pUrl.hostname}
+      const proxySocket = new net.Socket()
+      proxySocket.connect(connectOpts, () => {
+        proxySocket.write(`${req.method} ${pUrl.path} HTTP/1.1\r\n`)
+        proxySocket.write(`Host: ${apiUrl.host}\r\n`)
+        for(let i = 0; i < req.rawHeaders.length; i++) {
+          const key = req.rawHeaders[i]
+          if (key !== "Host" && key !== "Authorization") {
+            proxySocket.write(`${req.rawHeaders[i]}: ${req.rawHeaders[i+1]}\r\n`)
+          }
+          i++
+        }
```

Comment:
This is very error prone. I would recommend using `for (const [headerName, headerValue] of Object.entries(req.headers) { ... }`.

At the very least, having two places where you increment `i` seems like bad idea imo.

Reply:
@Nokel81 how would you improve this? `rawHeaders` is an array with paired values.

Reply:
Nevermind, you already posted an example how to improve 😄

Reply:
If you want to keep the handling of headers the same (since `headers` does some de-duplication. I would mearly change the loop to be:

```js
for (let i = 0; i < req.rawHeaders.length; i += 2) {
    ...
}
```

Reply:
Yes, we need rawHeaders. Thanks for the suggestion, much better now 👍